src/node_file.cc

```diff
@@ -149,8 +290,56 @@ void NewFSReqWrap(const FunctionCallbackInfo<Value>& args) {
   new FSReqWrap(env, args.This());
 }
 
+FSReqPromise::FSReqPromise(Environment* env)
+    : FSReqBase(env,
+                env->fsreqpromise_constructor_template()
+                    ->NewInstance(env->context()).ToLocalChecked(),
+                AsyncWrap::PROVIDER_FSREQPROMISE),
+      stats_field_array_(env->isolate(), 14) {
+  auto resolver = Promise::Resolver::New(env->context()).ToLocalChecked();
+  object()->Set(env->context(), env->promise_string(),
+                resolver.As<Promise>()).FromJust();
```

Comment:
@bmeurer ... just noting.... this is the bit we need to see if we can optimize (e.g. creating the promise in C++ using `Promise::Resolver::New`